plugins/modules/digital_ocean_domain.py

```diff
@@ -162,7 +162,8 @@ def core(module):
             if 'message' in domain:
                 module.fail_json(changed=False, msg=domain['message'])
             else:
-                module.exit_json(changed=True, domain=domain)
+                records = do_manager.all_domain_records()
+                module.exit_json(changed=True, domain=do_manager.domain_record())
```

Comment:
The [official docs](https://developers.digitalocean.com/documentation/v2/#create-a-new-domain) say this at the end:

> Keep in mind that, upon creation, the zone_file field will have a value of null until a zone file is generated and propagated through an automatic process on the DigitalOcean servers.

So this code just happens to work because listing all domain records and fetching the selected record once again gives the backend time to propagate the changes. Unfortunately, this method is as reliable as resolving race conditions by adding `sleep 3` to bash scripts. This will fail again if the DO backend becomes busy and takes more than a few seconds to update the zone file.

If we want to solve this reliably, we need to fetch data from API in a look until the zone file is present (with some upper bound of retries to avoid looping forever, of course).

Reply:
Ah, good find, I'll see what I can figure out.